crates/next-core/src/next_font_google/mod.rs

```diff
@@ -216,7 +221,44 @@ impl ImportMappingReplacement for NextFontGoogleCssModuleReplacer {
     }
 }
 
+#[turbo_tasks::function]
+async fn update_stylesheet(
+    stylesheet: StringVc,
+    options: NextFontGoogleOptionsVc,
+    scoped_font_family: StringVc,
+) -> Result<StringVc> {
+    // Update font-family definitions to the scoped name
+    // TODO: Do this more resiliently, e.g. transforming an swc ast
+    Ok(StringVc::cell(stylesheet.await?.replace(
+        &format!("font-family: '{}';", &*options.await?.font_family),
+        &format!("font-family: '{}';", &*scoped_font_family.await?),
+    )))
+}
+
+#[turbo_tasks::function]
+async fn get_scoped_font_family(query_vc: QueryMapVc) -> Result<StringVc> {
+    let options = font_options_from_query_map(query_vc).await?;
+
+    Ok(StringVc::cell(format!(
+        "__{}_{:x?}",
+        options.font_family.replace(' ', "_"),
+        *get_request_hash(query_vc).await? as u32
```

Comment:
Why the `as u32` here? Is this meant to truncate the hash? If so I'd rather `get_request_hash` itself do it with a comment explaining why.